File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map> 
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> map; 
    
    // 참가자 추가: C++에서는 맵에 없는 키에 접근할 때 기본값이 0으로 초기화된다 
    for(string name: participant){
        map[name]++; 
    }
    
    // 완주자 제거 
    for(string name: completion){
        map[name]--; 
    }
    
    // 값이 0보다 크면 완주하지 못한 사람임 
    for(auto entry: map){
        if(entry.second > 0){
            return entry.first; 
        }
    }
    return ""; 
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*; 
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        for(String s: participant){
            map.put(s, map.getOrDefault(s, 0)+1); 
        }
        
        for(String s: completion){
            map.put(s, map.get(s)-1); 
        }
        
        for(Map.Entry<String, Integer> e: map.entrySet()){
            if(e.getValue() > 0) return e.getKey();  
        }
        return answer;
    }
}
